frontend/src/components/categoryForm.js

```diff
@@ -0,0 +1,38 @@
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import InputField from "../components/input";
+
+const CategoryForm = ({ state, handleChange, handleSave, error }) => {
+  return (
+    <Form>
+      <InputField
+        name="title"
+        label="Title"
+        type="text"
+        onChange={handleChange}
+        value={state ? state.title : ""}
+        error={error ? error.title : null}
```

Comment:
short version, also apply to others
```suggestion
        value={state.title || ""}
        error={error.title || null}
```